bootstrap/kubeadm/api/v1alpha2/conversion.go

```diff
@@ -40,10 +40,23 @@ func (src *KubeadmConfig) ConvertTo(dstRaw conversion.Hub) error {
 	dst.Spec.Verbosity = restored.Spec.Verbosity
 	dst.Spec.UseExperimentalRetryJoin = restored.Spec.UseExperimentalRetryJoin
 	dst.Spec.Files = restored.Spec.Files
-	for i := range restored.Spec.Files {
-		file := restored.Spec.Files[i]
-		if file.ContentFrom != nil {
-			dst.Spec.Files = append(dst.Spec.Files, file)
+
+	// Track files successfully up-converted. We need this to dedupe
+	// restored files from user-updated files on up-conversion. We store
+	// them as pointers for later modification without paying for second
+	// lookup.
+	dstPaths := make(map[string]*kubeadmbootstrapv1alpha3.File, len(dst.Spec.Files))
+	for i := range dst.Spec.Files {
```

Comment:
```suggestion
	for _, file := range dst.Spec.Files {
```

Reply:
The first time you left this in review I agreed but for this one I don’t think I do because we need the pointer to the indexed value?

Reply:
yeah makes sense